lib/grandpa/grandpa.go

```diff
@@ -377,6 +377,7 @@ func (s *Service) initiate() error {
 
 func (s *Service) waitForFirstBlock() error {
 	ch := make(chan *types.Block)
+	// TODO (ed) import channel is unregistered by defer, doesn't seem to be closed
```

Comment:
I think this function has a problem. you can update the defer func to first `close(ch)` then call `Unregister()`, I think that should fix the issue at least partially, maybe there is a case where the GC collects `ch` and closes it before the `Unregister` is finished